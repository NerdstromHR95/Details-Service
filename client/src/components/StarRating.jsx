import React from 'react';

const StarRating = function(props) {
	let stars = [];
	const halfStar = (props.starRating % 20) !== 0
	const starCount = Math.floor(props.starRating / 20)
	for (let i = 0; i < starCount; i++) {
		stars.push(<i className="fas fa-star"></i>)
	}
	return (
		<div className='starRating'>
			{stars}
			{halfStar && <i className="fas fa-star-half"></i>}
		</div>
	)
}

export default StarRating;