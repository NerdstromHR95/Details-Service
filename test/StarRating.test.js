import React from 'react';
import StarRating from '../client/src/components/StarRating.jsx';
import renderer from 'react-test-renderer';

test('Should render 4 stars', () => {
	const component = renderer.create(<StarRating starRating={'80'} />);
	let tree = component.toJSON();

	expect(tree).toMatchSnapshot();
})