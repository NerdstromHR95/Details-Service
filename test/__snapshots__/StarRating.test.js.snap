// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 4 stars 1`] = `
<div
  className="starRating"
>
  <i
    className="fas fa-star"
  />
  <i
    className="fas fa-star"
  />
  <i
    className="fas fa-star"
  />
  <i
    className="fas fa-star"
  />
</div>
`;
